### ISSUE TITLE
FR: Scan things other that deployments

### ISSUE THREAD

**Issue by pastukhov** [stefanprodan/kubectl-kubesec#2]
Any plans to implement scannning kubernetes things  such as statefulsets or daemonsets?

**Comment by stefanprodan** [comment]
This needs to be implemented in kubesec first, pinging @sublimino

**Comment by sublimino** [comment]
OK, more tests added for failing cases and suite passing. To test everything in a cluster:

```
export -f kubesec
kubectl get pods,daemonset,statefulsets,deployments --all-namespaces -o=custom-columns=TYPE:.kind,NAME:.metadata.name,RSRC:.metadata.namespace --no-headers  \
  | awk '{print tolower($1)"/"$2" --namespace "$3}' \
  | xargs -I{} bash -xc 'TMP_FILE=$(mktemp); kubectl get -o yaml {} > ${TMP_FILE}; kubesec ${TMP_FILE};'
```

Should be all good for stateful/daemon sets now :D

**Comment by stefanprodan** [comment]
@pastukhov I've published v0.2, the command has changed to:
```
kubectl plugin scan [pod|deployment|statefulset|daemonset]/name
``` 

Example:

```
kubectl -n kube-system plugin scan daemonset/metadata-proxy-v0.1
scanning daemonset metadata-proxy-v0.1
daemonset/metadata-proxy-v0.1 kubesec.io score -32
-----------------
Critical
1. containers[] .securityContext .privileged == true
Privileged containers can allow almost completely unrestricted host access
2. .spec .hostNetwork
Sharing the host's network namespace permits processes in the pod to communicate with processes bound to the host's loopback adapter
-----------------
Advise
1. containers[] .securityContext .runAsNonRoot == true
Force the running image to run as a non-root user to ensure least privilege
2. containers[] .securityContext .capabilities .drop
Reducing kernel capabilities available to a container limits its attack surface
3. containers[] .securityContext .readOnlyRootFilesystem == true
An immutable root filesystem can prevent malicious binaries being added to PATH and increase attack cost
4. containers[] .securityContext .runAsUser > 10000
Run as a high-UID user to avoid conflicts with the host's user table
5. containers[] .securityContext .capabilities .drop | index("ALL")
Drop all capabilities and add only those required to reduce syscall attack surface
```

Please let me know if it works for you. Thanks
